packages/foam-vscode/src/utils.ts

```diff
@@ -184,3 +188,55 @@ export async function focusNote(notePath: string, moveCursorToEnd: boolean) {
     editor.selection = new Selection(range.end, range.end);
   }
 }
+
+export function getContainsTooltip(notes: Note[]): string {
+  const titles = notes.map(n => n.title).slice(0, 5);
+  return `Contains "${titles.join('", "')}"`;
+}
+
+/**
+ * Depending on the current vscode version, returns a MarkdownString of the
+ * note content casted as string or returns a simple string
+ * MarkdownString is only available from 1.52.1 onwards
+ * https://code.visualstudio.com/updates/v1_52#_markdown-tree-tooltip-api
+ * @param note A Foam Note
+ */
+export function getNoteTooltip(note: Note): string {
+  const STABLE_MARKDOWN_STRING_API_VERSION = '1.52.1';
+
+  if (version >= STABLE_MARKDOWN_STRING_API_VERSION) {
+    return formatMarkdownTooltip(note) as any;
+  }
+
+  return formatSimpleTooltip(note);
+}
+
+export function formatMarkdownTooltip(note: Note): MarkdownString {
+  const LINES_LIMIT = 15;
+  const markdown = firstNLines(stripFrontMatter(note.source.text), LINES_LIMIT);
+  return new MarkdownString(markdown);
+}
+
+export function formatSimpleTooltip(note: Note): string {
+  const WORDS_LIMIT = 30;
+  const words = removeMarkdown(stripFrontMatter(note.source.text))
+    .replace(/\r?\n|\r/g, ' ')
+    .replace(/\s+/g, ' ')
+    .replace(note.title, '')
```

Comment:
what do you want to achieve by removing the title?

Reply:
I thought it felt redundant as the mouse is hovering over it anyway, so it made sense to remove the first instance of the title. But we can keep it for sure, it's only for the tooltips < 1.52.1 anyway.

Reply:
yeah let's keep it simple, there might be some odd edge cases here